django/db/models/options.py

```diff
@@ -234,7 +235,7 @@ def _prepare(self, model):
                 if already_created:
                     field = already_created[0]
                 field.primary_key = True
-                self.setup_pk(field)
+                self.setup_pk(field, promoting_parent=True)
```

Comment:
hmm... I don't think that `auto_created` is relevant here, just `parent_link`, or am I missing something?